perf/src/bin/perf_client.rs

```diff
@@ -223,7 +257,7 @@ async fn drain_stream(mut stream: quinn::RecvStream, stats: &mut RequestStats) -
 async fn drive_uni(
     connection: quinn::Connection,
     acceptor: UniAcceptor,
-    stats: Arc<Mutex<Stats>>,
+    stream_stats: Arc<Mutex<Vec<Arc<StreamStats>>>>,
```

Comment:
Now that we're passing it around wrapped, I'd prefer that we abstract over the wrapping in a named type. (So some type potentially still named `StreamStats` should contain `Arc<Mutex<_>>` in a field.)